### ISSUE TITLE
Hotreload not working in .cshtml files while debugging UNLESS the project configuration is named "Debug"

### ISSUE THREAD

**Issue by alex-jitbit** [dotnet/aspnetcore#43910]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I've spent MONTHS investigating this. Finally found the cause.

My project configurations are named differently than the default "Debug/Release"

```xml
<Configurations>Full;Trial;FullCompany;FullSmall;FullDebug</Configurations>
<!--skipped some code-->
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='FullDebug|AnyCPU'">
	<DefineConstants>DEBUG;TRACE;FULL</DefineConstants>
</PropertyGroup>


```

And hot reload in Visual Studio is not working while debugging (F5), **unless my configuration is literally named "Debug"**.

### Expected Behavior

Should work

### .NET Version

6.0

**Comment by javiercn** [comment]
@alex-jitbit thanks for contacting us.

You might get a better and faster answer if you report this issue through the [Report a problem with Visual Studio - Visual Studio (Windows) | Microsoft Docs](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022) UI

**Comment by tmat** [comment]
Repro: 
[ConsoleApp12.zip](https://github.com/dotnet/aspnetcore/files/10775968/ConsoleApp12.zip)

**Comment by tmat** [comment]
@alex-jitbit Could you double-check that the following properties are set like so in your custom configuration?

```
    <Optimize>false</Optimize>
    <DebugSymbols>true</DebugSymbols>
```

**Comment by alex-jitbit** [comment]
@tmat yes they are, and like I said renaming the config helps

**Comment by tmat** [comment]
Can you share the project privately? What kind of project is it?

**Comment by tmat** [comment]
Never mind, I have a repro with Web App: 
[Repro.zip](https://github.com/dotnet/aspnetcore/files/10828490/Repro.zip)

**Comment by MaxenceMouchard** [comment]
Hello ! 
I opened a ticket on visual studio about this because I have the same problem.
This should be fixed in a future release according to the support team
[Ticket here](https://developercommunity.visualstudio.com/t/Hot-reloading-not-work-as-expected-when/10270598)

**Comment by MaxenceMouchard** [comment]
Has been fixed today, this ticket should be able to be closed

**Comment by BluMichele** [comment]
The issue persists in Visual Studio 2022 17.8

**Comment by mkArtakMSFT** [comment]
@BluMichele can you provide a repro please, because we thought this was fixed (at least the original report).

**Comment by MaxenceMouchard** [comment]
> The issue persists in Visual Studio 2022 17.8

Yes sorry for my enthusiasm it seems that the problem was corrected for a short time but it came back. The original report has not been fixed. I never responded positively to the ticket ... I don't know why this ticket has been tag as fixed.
Personally, I no longer run visual studio with a configuration named other than "Debug" du to a project infra changement.
So I don't have any more problems.

You should reopen an issue about this.

**Comment by MaxenceMouchard** [comment]
i'm on vs 17.7.4, the issue look fixed.
I've init a project with simple configuration if you want to test.
https://github.com/MaxenceMouchard/VSProjectHotReloadIssue.git

**Comment by miguno** [comment]
What's the latest status? Is it fixed by now or not in Visual Studio 2022?

**Comment by tmat** [comment]
As far as I can tell the issue is fixed.

@BluMichele Please open a new issue with a repro if it still doesn't work for you.
